booster/library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -84,8 +84,15 @@ throw = EquationT . lift . throwE
 catch_ ::
```

Comment:
I think this function now deserves a comment explaining what it does and why. How about we write something like: "Recover from an `EquationFailure`, taking into account the accumulated `EquationState` while executing the exception handler"?